src/osiris.erl

```diff
@@ -133,13 +134,14 @@ start_replica(Replica, Config) ->
 write(Pid, WriterId, Corr, Data) ->
     osiris_writer:write(Pid, self(), WriterId, Corr, Data).
 
--spec write_tracking(pid(), binary(), offset()) -> ok.
-write_tracking(Pid, TrackingId, Offset) ->
-    osiris_writer:write_tracking(Pid, TrackingId, Offset).
+-spec write_tracking(pid(), binary(), {tracking_type(), offset() | timestamp()}) -> ok.
+write_tracking(Pid, TrackingId, {_TrkType, _TrkData} = Tracking) ->
+    osiris_writer:write_tracking(Pid, TrackingId, Tracking).
 
--spec read_tracking(pid(), binary()) -> {tracking_type(), offset()} | undefined.
-read_tracking(Pid, TrackingId) ->
-    osiris_writer:read_tracking(Pid, TrackingId).
+-spec read_tracking(pid(), tracking_type(), binary()) ->
+    {tracking_type(), offset() | timestamp()} | undefined.
+read_tracking(Pid, TrackingType, TrackingId) ->
```

Comment:
This will also break code in rabbit. Perhaps we should retain backwards compat  by handling the previous call expectations.

Reply:
Pushed a commit to retain backwards compatibility.